plugins/chat/app/jobs/regular/auto_join_channel_batch.rb

```diff
@@ -30,9 +30,11 @@ def execute(args)
         last_seen_at: 3.months.ago,
         channel_category: channel.chatable_id,
         mode: UserChatChannelMembership.join_modes[:automatic],
+        permission_type: CategoryGroup.permission_types[:create_post],
       }
 
-      new_member_ids = DB.query_single(create_memberships_query(category), query_args)
+      restricted = join_restricted?(category)
```

Comment:
nit, but I don't think there is much point defining this here, just do it inside  `create_memberships_query`

Reply:
See other comment, can just get rid of restricted I think.